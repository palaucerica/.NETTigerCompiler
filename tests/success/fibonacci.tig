let
	type arr_int = array of int

	function fibonacci (n : int):int = 
	if n = 0 | n = 1 
	then 1
	else fibonacci(n - 1) + fibonacci(n - 2)        

	var values := arr_int[10] of 0
in
	values[0] := 2;
	values[1] := 4;
	values[2] := 6;
	values[3] := 8;
	values[4] := 10;
	values[5] := 12;
	values[6] := 14;
	values[7] := 16;
        values[8] := 18;
	values[9] := 20;
	for i := 0 to 9 do (
	    printi(values[i]);
	    print(" : ");
	    printi(fibonacci(values[i]));
	    print("\n"))
end