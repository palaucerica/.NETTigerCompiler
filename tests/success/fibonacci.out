2 : 2
4 : 5
6 : 13
8 : 34
10 : 89
12 : 233
14 : 610
16 : 1597
18 : 4181
20 : 10946
